metric/registry.go

```diff
@@ -53,8 +53,9 @@ func (r *Registry) AddInt64Gauge(name, description string, unit metricdata.Unit,
 }
 
 func (r *Registry) initGauge(g *gauge, labelKeys []string, name string, description string, unit metricdata.Unit) *gauge {
-	existing, ok := r.gauges[name]
+	val, ok := r.gauges.Load(name)
 	if ok {
+		existing := val.(*gauge)
 		if existing.isFloat != g.isFloat {
```

Comment:
As we discussed we need to return an error if it exists not only if it has a different type. But happy to see that in a different PR.

Reply:
I will create a different PR for this.